packages/react-dom/src/__tests__/ReactDOMComponentTree-test.js

```diff
@@ -68,56 +63,122 @@ describe('ReactDOMComponentTree', () => {
     expect(renderAndGetRef('input')).toBe('INPUT');
   });
 
-  it('finds instances for nodes', () => {
-    class Component extends React.Component {
+  it('finds closest instance for node when an event happens', () => {
+    const nonReactElemID = 'aID';
+    const innerHTML = {__html: `<div id="${nonReactElemID}"></div>`};
+    const closestInstanceID = 'closestInstance';
+    let currentTargetID = null;
+
+    class ClosestInstance extends React.Component {
+      _onClick = e => {
+        currentTargetID = e.currentTarget.id;
+      };
       render() {
         return (
-          <div>
-            <h1>hello</h1>
-            <p>
-              <input />
-            </p>
-            goodbye.
-            <main dangerouslySetInnerHTML={{__html: '<b><img></b>'}} />
-          </div>
+          <div
+            id={closestInstanceID}
+            onClick={this._onClick}
+            dangerouslySetInnerHTML={innerHTML}
+          />
         );
       }
     }
 
-    function renderAndQuery(sel) {
-      var root = renderMarkupIntoDocument(
-        <section>
-          <Component />
-        </section>,
-      );
-      return sel ? root.querySelector(sel) : root;
+    function simulateClick(elem) {
+      const event = new MouseEvent('click', {
+        bubbles: true,
+      });
+      elem.dispatchEvent(event);
     }
 
-    function renderAndGetInstance(sel) {
-      return ReactDOMComponentTree.getInstanceFromNode(renderAndQuery(sel));
+    const component = <ClosestInstance />;
+    ReactDOM.render(<section>{component}</section>, container);
+    expect(currentTargetID).toBe(null);
+    simulateClick(document.getElementById(nonReactElemID));
+    expect(currentTargetID).toBe(closestInstanceID);
+  });
+
+  it('finds a controlled instance from node and gets its current fiber props', () => {
```

Comment:
If you want to test `getFiberCurrentPropsFromNode`, I think there are more assertions you might need to make. For example I can currently replace its implementation with

```js
return node[internalInstanceKey].memoizedProps;
```

and the test still passes. However, this is a wrong implementation (`node[internalInstanceKey]` can potentially point to an "alternate" fiber that is not "current". Current and alternate swap places on updates.)

Can we make this test more solid by tracing what happens with "current" props? e.g. maybe they're used in some way where alternate props would've been wrong. You can git blame when `getFiberCurrentPropsFromNode` was added to maybe find the cases that were broken before it.

Reply:
Thanks for the detail, I'll check out the blame and track down ways to make the test more robust.